### PR TITLE
:sparkles:  Switch zap log for global hub agent

### PR DESCRIPTION
**Pull request by yanmxa**
## Related issue(s)

Fixes #  https://github.com/stolostron/multicluster-global-hub/issues/1178,  https://issues.redhat.com/browse/ACM-13421

1. Configure the LogLevel at runtime!
2. Integrate the log format with the third part library - **cloudevents**,  and **controller-runtime**
4. The underlayer is based on Zap `Logger` and `SugarLogger`, we also provide the `zapr` if you use the `logr` interface.
5. The manifestwork only updates the configmap, the agent deployment will not be updated. Let the operator read the config and propagate it to the configmap in the manifest
6. Format
```
2024-10-29T03:43:41.290Z        INFO    utils/utils.go:28       Go OS/Arch: linux/amd64
2024-10-29T03:43:41.290Z        INFO    utils/utils.go:29       Go Version: go1.22.5
2024-10-29T03:43:41.290Z        INFO    utils/utils.go:30       Git Commit: b46e471c
2024-10-29T03:43:41.313Z        INFO    controller-runtime.metrics      server/server.go:208    Starting metrics server
I1029 03:43:41.313460       1 leaderelection.go:254] attempting to acquire leader lease multicluster-global-hub-agent/multicluster-global-hub-agent-lock...
2024-10-29T03:43:41.313Z        INFO    controller-runtime.metrics      server/server.go:247    Serving metrics server  {"bindAddress": "0.0.0.0:8384", "secure": false}
I1029 03:43:41.317722       1 leaderelection.go:268] successfully acquired lease multicluster-global-hub-agent/multicluster-global-hub-agent-lock
2024-10-29T03:43:41.318Z        INFO    controller/controller.go:175    Starting EventSource    {"controller": "secret", "controllerGroup": "", "controllerKind": "Secret", "source": "kind source: *v1.Secret"}
2024-10-29T03:43:41.318Z        INFO    controller/controller.go:183    Starting Controller     {"controller": "secret", "controllerGroup": "", "controllerKind": "Secret"}
2024-10-29T03:43:41.521Z        INFO    controller/controller.go:217    Starting workers        {"controller": "secret", "controllerGroup": "", "controllerKind": "Secret", "worker count": 1}
2024-10-29T03:43:41.521Z        INFO    kafka-consumer  consumer/generic_consumer.go:84 transport consumer with cloudevents-kafka receiver
2024-10-29T03:43:41.525Z        INFO    cloudevents     v2@v2.0.0-20240911135016-682f3a9684e4/protocol.go:166   Subscribing to topics: [gh-spec]
2024-10-29T03:43:41.526Z        INFO    cmd/main.go:240 add the init controller to manager

```

We can see the cloudevents, controller runtime is the same as the format we set. But the lease still uses the klog, which we won't update in this PR. Visit here for more detail about that: https://github.com/kubernetes-sigs/controller-runtime/issues/2656, https://github.com/kubernetes/enhancements/issues/3077
## Tests
* [ ] Unit/function tests have been added and incorporated into `make unit-tests`.
* [ ] Integration tests have been added and incorporated into `make integration-test`.
* [ ] E2E tests have been added and incorporated into `make e2e-test-all`.
* [x] List other manual tests you have done.

### CODE REVIEW

**Comment by yanmxa** [comment]
e2e install error, retest

/test test-e2e

**Comment by yanmxa** [comment]
the environment isn't ready, re run it

/test e2e-test

**Comment by yanmxa** [comment]
setup error

/test test-e2e

**Comment by yanmxa** [comment]
database isn't ready

/test test-e2e

**Comment by yanmxa** [comment]
the kafka cluster is not created

/test test-e2e

File: operator/pkg/controllers/agent/manifests/templates/agent/multicluster-global-hub-agent-configmap.yaml [hunk 1]
```diff
@@ -11,4 +11,5 @@ data:
   hubClusterInfo: "60s"
   hubClusterHeartbeat: {{.AgentHeartbeatInteval}}
   aggregationLevel: {{ .AggregationLevel }}
-  enableLocalPolicies: "{{ .EnableLocalPolicies }}"
\ No newline at end of file
+  enableLocalPolicies: "{{ .EnableLocalPolicies }}"
+  logLevel: {{.LogLevel}}
```
Comment (clyang82) [inline operator/pkg/controllers/agent/manifests/templates/agent/multicluster-global-hub-agent-configmap.yaml:15]: where is `LogLevel` from? I do not see any changes in agent controller to set this value.

**Review by clyang82 — COMMENTED** [review]
Need to have a document about how to change the log level in the runtime environment

File: operator/pkg/controllers/agent/manifests/templates/agent/multicluster-global-hub-agent-configmap.yaml (hunk 1 above)
Comment (yanmxa) [reply operator/pkg/controllers/agent/manifests/templates/agent/multicluster-global-hub-agent-configmap.yaml:15]: The LogLevel is from the operator

The agent updates the LogLevel here: https://github.com/stolostron/multicluster-global-hub/pull/1184/files#diff-1f7a85f8429524557f22db6d08a8d0e48634c4073149d0eee6cc9a821d0e7d77R71-R75

**Comment by yanmxa** [comment]
> Need to have a document about how to change the log level in the runtime environment

Add the document in this pr: https://github.com/stolostron/multicluster-global-hub/pull/1188

**Pull request merged**